### ISSUE TITLE
 Failed to rasterize a picture: unable to create render target at specified size. exception on Navigator.pop

### ISSUE THREAD

**Issue by giaur500** [flutter/flutter#121485]
Android 11, Oneplus 9 Pro. Sometimes, after `Naviagator.pop` call, Flutter throws an exception:
```
I/flutter (25931): Failed to rasterize a picture: unable to create render target at specified size.
I/flutter (25931): #0      Canvas.drawImageRect (dart:ui/painting.dart:5174)
I/flutter (25931): #1      _drawImageScaledAndCentered (package:flutter/src/material/page_transitions_theme.dart:760)
I/flutter (25931): #2      _ZoomExitTransitionPainter.paintSnapshot (package:flutter/src/material/page_transitions_theme.dart:930)
I/flutter (25931): #3      _RenderSnapshotWidget.paint (package:flutter/src/widgets/snapshot_widget.dart:351)
I/flutter (25931): #4      RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2853)
I/flutter (25931): #5      PaintingContext.paintChild (package:flutter/src/rendering/object.dart:253)
I/flutter (25931): #6      RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:144)
I/flutter (25931): #7      _ZoomEnterTransitionPainter.paint (package:flutter/src/material/page_transitions_theme.dart:866)
I/flutter (25931): #8      _RenderSnapshotWidget.paint (package:flutter/src/widgets/snapshot_widget.dart:335)
I/flutter (25931): #9      RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2853)
I/flutter (25931): #10     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:253)
I/flutter (25931): #11     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:144)
I/flutter (25931): #12     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2853)
I/flutter (25931): #13     PaintingContext._repaintCompositedChild (package:flutter/src/rendering/object.dart:169)
I/flutter (25931): #14     PaintingContext.repaintCompositedChild (package:flutter/src/rendering/object.dart:112)
I/flutter (25931): #15     PipelineOwner.flushPaint (package:flutter/src/rendering/object.dart:1137)
I/flutter (25931): #16     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:518)
I/flutter (25931): #17     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:865)
I/flutter (25931): #18     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:381)
I/flutter (25931): #19     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1289)
I/flutter (25931): #20     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1218)
I/flutter (25931): #21     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1076)
I/flutter (25931): #22     _invoke (dart:ui/hooks.dart:145)
I/flutter (25931): #23     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:338)
I/flutter (25931): #24     _drawFrame (dart:ui/hooks.dart:112)

```
It seems nothing else happens, app does not crash. It's completly random and also I am not able to reproduce it on Android emulator in debug mode.

Additionally, sometimes I'm also getting a few exceptions like that:

```
I/flutter (25931): Another exception was thrown: Instance of 'DiagnosticsProperty<void>'
I/flutter (25931): Another exception was thrown: Instance of 'DiagnosticsProperty<void>'
I/flutter (25931): Another exception was thrown: Instance of 'DiagnosticsProperty<void>'
I/flutter (25931): Another exception was thrown: Instance of 'DiagnosticsProperty<void>'
```

In this case, app does not crash too, but transition does not work at all, I'm getting black screen for a while and no any transition while pop page.

```
flutter doctor -v
[✓] Flutter (Channel stable, 3.7.5, on Debian GNU/Linux bookworm/sid 6.1.0-10.slh.1-aptosid-amd64, locale pl_PL.UTF-8)
    • Flutter version 3.7.5 on channel stable at /home/michal2/Android/flutter/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c07f788888 (4 dni temu), 2023-02-22 17:52:33 -0600
    • Engine revision 0f359063c4
    • Dart version 2.19.2
    • DevTools version 2.20.1

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/michal2/Android/Sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/linux#android-setup for more details.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • Debian clang version 14.0.6
    • cmake version 3.25.1
    • ninja version 1.11.1
    • pkg-config version 1.8.1

[✓] Android Studio (version 4.2)
    • Android Studio at /home/michal2/android/studio
    • Flutter plugin version 57.0.2
    • Dart plugin version 202.8488
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)

[✓] IntelliJ IDEA Community Edition (version 2019.2)
    • IntelliJ at /home/michal2/Pobrane/intelij/ideaIC-2019.2/idea-IC-192.5728.98
    • Flutter plugin version 38.2.4
    • Dart plugin version 192.6262.13

[✓] Connected device (3 available)
    • LE2120 (mobile) • eaef19cf • android-arm64  • Android 11 (API 30)
    • Linux (desktop) • linux    • linux-x64      • Debian GNU/Linux bookworm/sid 6.1.0-10.slh.1-aptosid-amd64
    • Chrome (web)    • chrome   • web-javascript • Google Chrome 110.0.5481.77

[✓] HTTP Host Availability
    • All required HTTP hosts are available

```

**Comment by danagbemava-nc** [comment]
Hi @giaur500,

We'll need a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) so that we can investigate this. How often does the crash reproduce & on what devices are you experiencing it?

Thank you

**Comment by giaur500** [comment]
Hello,
The problem is that it's only sometimes and also in release mode only, so it's hard to debug. Aslo, it's only one specific (complicated) scenario. I've tried with clean project with simple 2 pages and I am not able to reproduce it.

Could you provide any advice how to investiagete it deeper? I'm out of ideas now. It started happening after update from Flutter 2.5.3 to 3.7.5.

Problematic page is pushed like this (no null safety for now):

 ```
Navigator.push(
        context,
        MaterialPageRoute(
          builder: (context) => PlacePicker(
            onPlacePicked: (result) async {
            // .....
              Navigator.of(context).pop();

              if (await Navigator.push(
                context,
                MaterialPageRoute(
                  builder: (context) => AddEventPage.withData(
                   // ....
                  ),
                ),
              ) ??
                  false) _someOtherTask();
            },
          ),
        ),
      );

```
So, first `PlacePicker` page needs to be displayed, after address picked, `Navigator.of(context).pop();` call to close picker window and then, open `AddEventPage`. Execption occurs on `AddEventPage` pop - it includes Cancel button and to pop page `Navigator.of(context).pop(false);` or `Navigator.of(context).pop(true)` for Ok button to run `_someOtherTask();`

Do you have any idea what could cause exceptions like that? I also will keep trying to reproduce it in simple app, but it's hard to do.

**Comment by jonahwilliams** [comment]
FYI @dnfield . maybe we should update the error to include the render target size? Its possible some combination of things going wrong leads to an absurdly large texture. Without a repro it will be difficult to impossible to fix.

**Comment by danagbemava-nc** [comment]
> Do you have any idea what could cause exceptions like that?

Unfortunately, I do not. Are you by any chance using custom page transitions?

> I also will keep trying to reproduce it in a simple app

Please do and keep up posted

Thank you

**Comment by giaur500** [comment]
> Hi @giaur500, unfortunately, I do not. Are you by any chance using custom page transitions?

I'm actually not forced to use default transition, do you think implementing custom one may help?

**Comment by danagbemava-nc** [comment]
> I'm actually not forced to use default transition, do you think implementing custom one may help

It could, but I was only asking just in case you were already using one and perhaps some API that you depended had changed in ways that were not expected. 

> Also, is there any way to get more detailed stack trace to help to track the issue? In release mode, all I can do is flutter logs from terminal + dump error to console in my code.

Since this is occurring in only release mode, I think you should be able to use some crash monitoring tool like crashlytics or sentry which should upload the crashes to somewhere you can monitor. Sentry can record the routes in your app and I imagine crashlytics should be able to do same

**Comment by giaur500** [comment]
@danagbemava-nc Crashlytics only gives the same stack trace as I provided before:

```
Non-fatal Exception: io.flutter.plugins.firebase.crashlytics.FlutterError: Failed to rasterize a picture: unable to create render target at specified size.. Error thrown Instance of 'ErrorDescription'.
       at Canvas.drawImageRect(Canvas.java)
       at ._drawImageScaledAndCentered(page_transitions_theme.dart:760)
       at _ZoomExitTransitionPainter.paintSnapshot(page_transitions_theme.dart:930)
       at _RenderSnapshotWidget.paint(snapshot_widget.dart:351)
       at RenderObject._paintWithContext(object.dart:2853)
       at PaintingContext.paintChild(object.dart:253)
       at RenderProxyBoxMixin.paint(proxy_box.dart:144)
       at _ZoomEnterTransitionPainter.paint(page_transitions_theme.dart:866)
       at _RenderSnapshotWidget.paint(snapshot_widget.dart:335)
       at RenderObject._paintWithContext(object.dart:2853)
       at PaintingContext.paintChild(object.dart:253)
       at RenderProxyBoxMixin.paint(proxy_box.dart:144)
       at RenderObject._paintWithContext(object.dart:2853)
       at PaintingContext._repaintCompositedChild(object.dart:169)
       at PaintingContext.repaintCompositedChild(object.dart:112)
       at PipelineOwner.flushPaint(object.dart:1137)
       at RendererBinding.drawFrame(binding.dart:518)
       at WidgetsBinding.drawFrame(binding.dart:865)
       at RendererBinding._handlePersistentFrameCallback(binding.dart:381)
       at SchedulerBinding._invokeFrameCallback(binding.dart:1289)
       at SchedulerBinding.handleDrawFrame(binding.dart:1218)
       at SchedulerBinding._handleDrawFrame(binding.dart:1076)
```
There will be nothig more, unless anything more will be generated. To record exceptions:

 ```
FlutterError.onError = (FlutterErrorDetails details) {
    FlutterError.dumpErrorToConsole(details);

    if (!details.silent) FirebaseCrashlytics.instance.recordFlutterError(details);

    /*if (!details.silent && kReleaseMode) {
      exit(1);
    }*/
  };
```
So, crashlytics and `flutter logs` will always show the same stacktrace. I am forced to disable `exit(1)` on crash, because that issue casues app to exit unexpectedly - exception is not silent, but it actually does not cause any crash visible for user (except no transition sometimes).

Maybe it's possible to fork/subclass default zoom transition and apply more logging there?

**Comment by asaarnak** [comment]
@danagbemava-nc @jonahwilliams 
Same exception appeared for us since Flutter 3.7.0.
We use Navigator 2.0.
Exception appeared randomly, about 3/10 times on cheaper phones when navigating pages very fast.
![image](https://user-images.githubusercontent.com/808420/222090954-684f9014-689f-4b2d-a249-f7ac6fbddb7c.png)

Managed to reproduce it locally on emulator using older OpenGL:
![image](https://user-images.githubusercontent.com/808420/222110045-f57d2c1b-4d3c-4feb-89f5-45373564df37.png)
<details>
  <summary>Emulator logs and stacktrace in debug mode</summary>
  
```stacktrace
D/MapsInitializer( 8218): preferredRenderer: null
D/zzcb    ( 8218): preferredRenderer: null
I/Google Maps Android API( 8218): Google Play services package version: 225014047
I/Google Maps Android API( 8218): Google Play services maps renderer version(legacy): 203115000
I/PlatformViewsController( 8218): Using hybrid composition for platform view: 24
E/GoogleMapController( 8218): Cannot enable MyLocation layer as location permissions are not granted
D/MapsInitializer( 8218): preferredRenderer: null
D/zzcb    ( 8218): preferredRenderer: null
I/Google Maps Android API( 8218): Google Play services package version: 225014047
I/Google Maps Android API( 8218): Google Play services maps renderer version(legacy): 203115000
I/PlatformViewsController( 8218): Using hybrid composition for platform view: 25
E/GoogleMapController( 8218): Cannot enable MyLocation layer as location permissions are not granted
E/libEGL  ( 8218): call to OpenGL ES API with no current context (logged once per thread)
D/EGL_emulation( 8218): app_time_stats: avg=179.06ms min=5.10ms max=1285.44ms count=8

======== Exception caught by rendering library =====================================================
The following PictureRasterizationException was thrown during paint():
Failed to rasterize a picture: unable to create render target at specified size.
The callstack when the image was created was:
#0      new Image._.<anonymous closure> (dart:ui/painting.dart:1670:32)
#1      new Image._ (dart:ui/painting.dart:1672:6)
#2      Scene.toImageSync (dart:ui/compositing.dart:34:18)
#3      OffsetLayer.toImageSync (package:flutter/src/rendering/layer.dart:1497:20)
#4      _RenderSnapshotWidget._paintAndDetachToImage (package:flutter/src/widgets/snapshot_widget.dart:315:40)
#5      _RenderSnapshotWidget.paint (package:flutter/src/widgets/snapshot_widget.dart:345:22)
#6      RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2853:7)
#7      PaintingContext.paintChild (package:flutter/src/rendering/object.dart:253:13)
#8      RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:144:15)
#9      _ZoomEnterTransitionPainter.paint (package:flutter/src/material/page_transitions_theme.dart:866:23)
#10     _RenderSnapshotWidget.paint (package:flutter/src/widgets/snapshot_widget.dart:335:15)
#11     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2853:7)
#12     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:253:13)
#13     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:144:15)
#14     _ZoomExitTransitionPainter.paint (package:flutter/src/material/page_transitions_theme.dart:938:23)
#15     _RenderSnapshotWidget.paint (package:flutter/src/widgets/snapshot_widget.dart:335:15)
#16     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2853:7)
#17     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:253:13)
#18     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:144:15)
#19     _ZoomEnterTransitionPainter.paint (package:flutter/src/material/page_transitions_theme.dart:866:23)
#20     _RenderSnapshotWidget.paint (package:flutter/src/widgets/snapshot_widget.dart:335:15)
#21     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2853:7)
#22     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:253:13)
#23     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:144:15)
#24     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2853:7)
#25     PaintingContext._repaintCompositedChild (package:flutter/src/rendering/object.dart:169:11)
#26     PaintingContext.repaintCompositedChild (package:flutter/src/rendering/object.dart:112:5)
#27     PipelineOwner.flushPaint (package:flutter/src/rendering/object.dart:1137:31)
#28     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:518:19)
#29     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:865:13)
#30     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:381:5)
#31     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1289:15)
#32     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1218:9)
#33     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1076:5)
#34     _invoke (dart:ui/hooks.dart:145:13)
#35     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:338:5)
#36     _drawFrame (dart:ui/hooks.dart:112:31)

The relevant error-causing widget was: 
  Navigator-[LabeledGlobalKey<NavigatorState>#89166] Navigator:file:///media/asaarnak/A/CLIENTS/northcoastcode/peppermint/peppermint/lib/dr_house/screens/home/home/make_an_appointment/virtual_visit_booking_route.dart:161:18
When the exception was thrown, this was the stack: 
#0      Canvas.drawImageRect (dart:ui/painting.dart:5174:7)
#1      _drawImageScaledAndCentered (package:flutter/src/material/page_transitions_theme.dart:760:18)
#2      _ZoomExitTransitionPainter.paintSnapshot (package:flutter/src/material/page_transitions_theme.dart:930:5)
#3      _RenderSnapshotWidget.paint (package:flutter/src/widgets/snapshot_widget.dart:351:15)
#4      RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2853:7)
#5      PaintingContext.paintChild (package:flutter/src/rendering/object.dart:253:13)
#6      RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:144:15)
#7      _ZoomEnterTransitionPainter.paint (package:flutter/src/material/page_transitions_theme.dart:866:23)
#8      _RenderSnapshotWidget.paint (package:flutter/src/widgets/snapshot_widget.dart:335:15)
#9      RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2853:7)
#10     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:253:13)
#11     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:144:15)
#12     _ZoomExitTransitionPainter.paint (package:flutter/src/material/page_transitions_theme.dart:938:23)
#13     _RenderSnapshotWidget.paint (package:flutter/src/widgets/snapshot_widget.dart:335:15)
#14     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2853:7)
#15     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:253:13)
#16     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:144:15)
#17     _ZoomEnterTransitionPainter.paint (package:flutter/src/material/page_transitions_theme.dart:866:23)
#18     _RenderSnapshotWidget.paint (package:flutter/src/widgets/snapshot_widget.dart:335:15)
#19     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2853:7)
#20     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:253:13)
#21     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:144:15)
#22     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2853:7)
#23     PaintingContext._repaintCompositedChild (package:flutter/src/rendering/object.dart:169:11)
#24     PaintingContext.repaintCompositedChild (package:flutter/src/rendering/object.dart:112:5)
#25     PipelineOwner.flushPaint (package:flutter/src/rendering/object.dart:1137:31)
#26     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:518:19)
#27     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:865:13)
#28     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:381:5)
#29     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1289:15)
#30     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1218:9)
#31     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1076:5)
#32     _invoke (dart:ui/hooks.dart:145:13)
#33     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:338:5)
#34     _drawFrame (dart:ui/hooks.dart:112:31)
The following RenderObject was being processed when the exception was fired: _RenderSnapshotWidget#50b80
...  needs compositing
...  parentData: <none> (can use size)
...  constraints: BoxConstraints(w=375.3, h=788.0)
...  size: Size(375.3, 788.0)
RenderObject: _RenderSnapshotWidget#50b80
  needs compositing
  parentData: <none> (can use size)
  constraints: BoxConstraints(w=375.3, h=788.0)
  size: Size(375.3, 788.0)
...  child: RenderIgnorePointer#d84e2
...    needs compositing
...    parentData: <none> (can use size)
...    constraints: BoxConstraints(w=375.3, h=788.0)
...    size: Size(375.3, 788.0)
...    ignoring: false
...    ignoringSemantics: implicitly false
...    child: RenderRepaintBoundary#f5ee9
...      needs compositing
...      parentData: <none> (can use size)
...      constraints: BoxConstraints(w=375.3, h=788.0)
...      layer: OffsetLayer#0df9b DETACHED
...        engine layer: OffsetEngineLayer#8c87f
...        handles: 1
...        offset: Offset(0.0, 0.0)
...      size: Size(375.3, 788.0)
...      metrics: 75.0% useful (1 bad vs 3 good)
...      diagnosis: insufficient data to draw conclusion (less than five repaints)
...      child: RenderSemanticsAnnotations#148fd
...        needs compositing
...        parentData: <none> (can use size)
...        constraints: BoxConstraints(w=375.3, h=788.0)
...        size: Size(375.3, 788.0)
...        child: RenderPhysicalModel#86a51
...          needs compositing
...          parentData: <none> (can use size)
...          constraints: BoxConstraints(w=375.3, h=788.0)
...          size: Size(375.3, 788.0)
...          elevation: 0.0
...          color: Color(0xfff8f4ee)
...          shadowColor: Color(0xfff8f4ee)
...          shape: BoxShape.rectangle
...          borderRadius: BorderRadius.zero
====================================================================================================
```

</details>

Error appears with default values for widgets:
1. MaterialPage: `this.allowSnapshotting = true,` 
2. GoogleMap: `GoogleMapsFlutterPlatform.instance.useAndroidViewSurface = true;`.

Workaround:
Using `allowSnapshotting = false,` or `useAndroidViewSurface = false` made this error go away.
We are now using `useAndroidViewSurface = false` since Flutter 3.7.3.

This error seems to be related to GPU context not available: https://api.flutter.dev/flutter/dart-ui/PictureRasterizationException-class.html because before stacktrace there was error log: `E/libEGL  ( 8218): call to OpenGL ES API with no current context (logged once per thread)`

**Comment by giaur500** [comment]
@asaarnak  where to apply `allowSnapshotting = false` or `useAndroidViewSurface = false` ?

**Comment by asaarnak** [comment]
1. We don't use `allowSnapshotting: false` but it could be set on MaterialPage widget, it could lead to slower performance.
2. We set `useAndroidViewSurface = false` in main.dart like this.
You need to add google_maps_flutter_android to pubspec.yaml.
```dart
final mapsImplementation = GoogleMapsFlutterPlatform.instance;
if (mapsImplementation is GoogleMapsFlutterAndroid) {
  // Related issues:
  // https://github.com/flutter/flutter/issues/103686
  // https://github.com/flutter/flutter/issues/121485
  // Default value will probably change in future
  mapsImplementation.useAndroidViewSurface = false;
  try {
    // Related issue: https://github.com/flutter/flutter/issues/105965
    await mapsImplementation.initializeWithRenderer(
      AndroidMapRenderer.latest,
    );
  } on PlatformException catch (exception) {
    if (exception.code == 'Renderer already initialized') {
      // Happens during hot reload
      return;
    }
    rethrow;
  }
}
```

**Comment by giaur500** [comment]
`GoogleMapsFlutterPlatform` is still undefined.  Can I do that this way?

```
 if (defaultTargetPlatform == TargetPlatform.android) {
    AndroidGoogleMapsFlutter.useAndroidViewSurface = false;
  }
```
(but it's deprecated)

**Comment by giaur500** [comment]
Yes, but still GoogleMapsFlutterPlatform is not available

**Comment by asaarnak** [comment]
@danagbemava-nc 
Created https://github.com/asaarnak/google_maps_test
Was quite easily reproduceable on emulator API Level 33, x86_64, but couln't reproduce on Note 20 Ultra.
Click Map and Home button to navigate between pages until error appears.

**Comment by danagbemava-nc** [comment]
Hi @asaarnak, just for confirmation, I noticed in your readme that you're using an invalid key, is that a requirement for reproducing the bug, or occurs even with a valid key?

**Comment by giaur500** [comment]
@danagbemava-nc In my case it occurs with valid key, I think you should test with valid key and working map. I also noticed another issue and it may be related - if you push map page and then pop it, pop animation is broken. Issues goes away with `useAndroidViewSurface = false`

**Comment by giaur500** [comment]
@asaarnak are you also getting this?

`I/flutter (25931): Another exception was thrown: Instance of 'DiagnosticsProperty<void>'`

**Comment by danagbemava-nc** [comment]
Hi @asaarnak & @giaur500, I was able to reproduce the error on the `stable` channel on a Nokia 6 but I cannot seem to reproduce the error on the `master` channel. Can either of you confirm if this is still reproducible on the `master` channel?

**Comment by giaur500** [comment]
> this seems like a separate issue, so you might have to file a new issue for that as well
It seem after applying asaarnak's workround both exceptions are gone, so they could be related. Actually, `Another exception was thrown: Instance of 'DiagnosticsProperty'` says nothing

**Comment by danagbemava-nc** [comment]
@giaur500, Thanks for the info. 

Have you been able to confirm if the bug reproduces on the `master` channel?

**Comment by harisijazwarraich** [comment]
Here's the stacktrace from logs.
```
painting.dart in Canvas.drawImageRect at line 5174
page_transitions_theme.dart in _drawImageScaledAndCentered at line 760 within flutter
page_transitions_theme.dart in _ZoomExitTransitionPainter.paintSnapshot at line 930 within flutter
snapshot_widget.dart in _RenderSnapshotWidget.paint at line 351 within flutter
object.dart in RenderObject._paintWithContext at line 2853 within flutter
object.dart in PaintingContext.paintChild at line 253 within flutter
layout_builder.dart in _RenderLayoutBuilder.paint at line 338 within flutter
page_transitions_theme.dart in _ZoomEnterTransitionPainter.paint at line 866 within flutter
snapshot_widget.dart in _RenderSnapshotWidget.paint at line 335 within flutter
object.dart in RenderObject._paintWithContext at line 2853 within flutter
object.dart in PaintingContext.paintChild at line 253 within flutter
layout_builder.dart in _RenderLayoutBuilder.paint at line 338 within flutter
object.dart in RenderObject._paintWithContext at line 2853 within flutter
object.dart in PaintingContext._repaintCompositedChild at line 169 within flutter
object.dart in PaintingContext.repaintCompositedChild at line 112 within flutter
object.dart in PipelineOwner.flushPaint at line 1137 within flutter
binding.dart in RendererBinding.drawFrame at line 518 within flutter
binding.dart in WidgetsBinding.drawFrame at line 865 within flutter
binding.dart in RendererBinding._handlePersistentFrameCallback at line 381 within flutter
binding.dart in SchedulerBinding._invokeFrameCallback at line 1289 within flutter
binding.dart in SchedulerBinding.handleDrawFrame at line 1218 within flutter
binding.dart in SchedulerBinding._handleDrawFrame at line 1076 within flutter
hooks.dart in _invoke at line 145
platform_dispatcher.dart in PlatformDispatcher._drawFrame at line 338
hooks.dart in _drawFrame at line 112
```

**Comment by giaur500** [comment]
@danagbemava-nc My project won't run on Flutter from master, so I'm not able to check, sorry.

**Comment by giaur500** [comment]
> This bug wouldn't cause an app crash, but it may mean that you're rendering looks wrong or glitchy.

Indeed, in my case it causes no transition and black screen instead. No app crash though, How is that related to Google Maps?

**Comment by danagbemava-nc** [comment]
Labeling for further investigation as I cannot conclusively say that this has been fixed on the master channel and I do not have the device I used to reproduce the bug for the time being. 

Steps to reproduce and code sample can be found in https://github.com/flutter/flutter/issues/121485#issuecomment-1450272432 

According to OP in https://github.com/flutter/flutter/issues/121485#issuecomment-1460546039, the workaround in https://github.com/flutter/flutter/issues/121485#issuecomment-1449809503 works for now.

**Comment by giaur500** [comment]
>  I do not have the device I used to reproduce the bug for the time being.

It's been mentioned issue can be reproduced on emulator, have you tried on emulator? Workaround works, but it causes other issues

**Comment by danagbemava-nc** [comment]
> It's been mentioned issue can be reproduced on emulator, have you tried on emulator? Workaround works, but it causes other issues

I have issues with using emulators on my device that makes it difficult for me to use them for triage. It slows my device to a halt making it difficult to do anything else. It's not much better after a restart either

**Comment by dnfield** [comment]
> > I do not have the device I used to reproduce the bug for the time being.
> 
> It's been mentioned issue can be reproduced on emulator, have you tried on emulator? Workaround works, but it causes other issues

It's much more interesting if the issue can be reproduced on a device. Emulators have all sorts of bugs around this kind of thing that we're likely never going to be able to fix (sometimes it's a bug in the GL translation stack on the emulator, for example).

**Comment by dnfield** [comment]
If this is happening with Google maps, it's also remotely possible that it's related to https://github.com/flutter/flutter/issues/120815

**Comment by dnfield** [comment]
Also, if you can try your reproduction again on the latest Flutter commit, you should get a _slightly_ more detailed error message.

**Comment by giaur500** [comment]
@dnfield
> It's much more interesting if the issue can be reproduced on a device. Emulators have all sorts of bugs around this kind of thing that we're likely never going to be able to fix (sometimes it's a bug in the GL translation stack on the emulator, for example).

It can for sure. I got it on Oneplus 9 Pro with on stock Android 11 and LG G5 with LineageOS. From other side, never got it on emulator. On my Linux emulator (x86-64 images) works very fast without any issues, no idea how does it work on Windows.

**Comment by giaur500** [comment]
> If this is happening with Google maps, it's also remotely possible that it's related to #120815
 
Isn't that Linux only issue? Linux is not even supported by Google Maps

**Comment by dnfield** [comment]
> > If this is happening with Google maps, it's also remotely possible that it's related to #120815
> 
> Isn't that Linux only issue? Linux is not even supported by Google Maps

Yes, but it's possible the FBO binding issue is causing this one as well :)

**Comment by dnfield** [comment]
There's a fair chance this will be fixed by https://github.com/flutter/engine/pull/40208.

I have not reproduced this yet, but here's what I think _could_ be happening:

- Platform view is forcing a thread merge
- Something is happening where the GL context is set to something the rasterizer doesn't know about
- We ask Skia to do work without making the context current and Skia corrupts the state of the other GL context
- Sadness

The linked patch makes sure that toImageSync tries to claim the right context before proceding.

**Comment by dnfield** [comment]
Can anyone who was able to reproduce this try on a recent commit from main/master to see if it still happens?

**Comment by asaarnak** [comment]
I'm on vacation, could try on 23 Mar.

**Comment by chinmaygarde** [comment]
From triage: Closing this as fixed. Please re-open if you are able to reproduce this after the recent commit.

**Comment by giaur500** [comment]
Will it be merged to Flutter 3 or Flutter 4 only?

**Comment by dnfield** [comment]
Check https://github.com/flutter/flutter/commit/c47daaa472a64b94c0389f93200524546512bd1d

It's currently available in the 3.9.0-6.0.pre and newer. That should mean it will eventually be available in 3.9.
